### PR TITLE
[Serve] Add more logging data for autoscaling in the log file & add request stats into panel

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

- Add replica queuing metrics.
- Add debug message for autoscaling. Enabled by setting SERVE_DEBUG_LOG=1
- Add replica running/pending tasks into the graffana. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/autoscaling_policy.py [hunk 1]
```diff
@@ -155,6 +171,11 @@ def get_decision_num_replicas(
             if self.decision_counter > self.scale_up_consecutive_periods:
                 self.decision_counter = 0
                 decision_num_replicas = desired_num_replicas
+                self._logging_autoscaling_metrics(
```
Comment (edoakes) [inline python/ray/serve/_private/autoscaling_policy.py:174]:
```suggestion
                self._log_autoscaling_metrics(
```

**Review by edoakes — COMMENTED** [review]
@sihanwang41 can you please also add `INFO` level logs for all important state transitions? i.e., when the autoscaler decides to scale up / scale down a deployment we should have a log statement for it (including relevant summary stats).

**Comment by sihanwang41** [comment]
> @sihanwang41 can you please also add `INFO` level logs for all important state transitions? i.e., when the autoscaler decides to scale up / scale down a deployment we should have a log statement for it (including relevant summary stats).

Hi @edoakes, currently `_log_autoscaling_metrics` function will only printed when autoscaler make a decision to change the num of replicas. including information about current num replicas & desired num replicas & qps from all replicas.

File: python/ray/serve/_private/autoscaling_policy.py [hunk 2]
```diff
@@ -123,6 +126,20 @@ def __init__(self, config: AutoscalingConfig):
...
+        self,
+        curr_target_num_replicas: int,
+        current_num_ongoing_requests: List[float],
+        desired_num_replicas: int,
+    ):
+        autoscaling_msg = (
+            "Autoscaling decision:"
+            f"curr_target_num_replicas:  {curr_target_num_replicas}, "
+            f"current_num_ongoing_requests: {current_num_ongoing_requests}, "
+            f"desired_num_replicas: {desired_num_replicas}"
```
Comment (edoakes) [inline python/ray/serve/_private/autoscaling_policy.py:139]: this should include the deployment name and let's make it more readable:
"Autoscaler updating num_replicas for deployment 'name' from <current> to <target>. current_num_ongoing_requests: ..."

File: python/ray/serve/_private/replica.py [hunk 3]
```diff
@@ -445,6 +445,11 @@ def user_health_check():
             description="The current number of queries being processed.",
         )
 
+        self.num_pending_items = metrics.Gauge(
+            "serve_replica_pending_queries",
+            description="The number of queries being pended.",
```
Comment (zcin) [inline python/ray/serve/_private/replica.py:450]:
```suggestion
            description="The current number of pending queries.",
```

**Review by alanwguo — APPROVED** [review]
panels lgtm

**Comment by sihanwang41** [comment]
@edoakes ^^ please take another look

**Review by edoakes — APPROVED** [review]

**Pull request merged**